### PR TITLE
chore: v2 api changes for session token endpoint

### PR DESCRIPTION
**Pull request by hrithikesh026**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Added the new route for v2 session token endpoint. And also generated the openapi spec file.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Hit this api.
```
curl --location 'http://localhost:8080/v2/payments/pay_123/create_external_sdk_tokens' \
--header 'Content-Type: application/json' \
--data '{
}'
```
Should get below response:
```
Session Response
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/api_models/src/payments.rs [hunk 1]
```diff
@@ -4787,6 +4800,17 @@ pub struct RedirectResponse {
     pub json_payload: Option<pii::SecretSerdeValue>,
 }
 
+#[cfg(all(feature = "v2", feature = "payment_v2"))]
+#[derive(Debug, serde::Deserialize, serde::Serialize, Clone, ToSchema)]
+pub struct PaymentsSessionRequest {
+    /// This is a token which expires after 15 minutes, used from the client to authenticate and create sessions from the SDK
+    pub client_secret: String,
```
Comment (Narayanbhat166) [inline crates/api_models/src/payments.rs:4807]: can you use `ClientSecret` type for this?

File: crates/api_models/src/payments.rs [hunk 2]
```diff
@@ -4746,6 +4761,15 @@ pub struct RedirectResponse {
     pub json_payload: Option<pii::SecretSerdeValue>,
 }
 
+#[cfg(feature = "v2")]
+#[derive(Debug, serde::Deserialize, serde::Serialize, Clone, ToSchema)]
+pub struct PaymentsSessionRequest {
+    /// This is a token which expires after 15 minutes, used from the client to authenticate and create sessions from the SDK
+    #[schema(value_type = String)]
+    pub client_secret: ClientSecret,
```
Comment (jarnura) [inline crates/api_models/src/payments.rs:4769]: Move client_secret to headers

File: crates/openapi/src/routes/payments.rs [hunk 3]
```diff
@@ -381,12 +381,31 @@ pub fn payments_confirm() {}
 )]
 pub fn payments_capture() {}
 
+#[cfg(feature = "v1")]
+/// Payments - Session token
+///
+/// Creates a session object or a session token for wallets like Apple Pay, Google Pay, etc. These tokens are used by Hyperswitch's SDK to initiate these wallets' SDK.
+#[utoipa::path(
+  post,
```
Comment (jarnura) [inline crates/openapi/src/routes/payments.rs:389]: Change to get since we don't have body for this

**Review by jarnura — CHANGES_REQUESTED** [review]

**Review by jarnura — APPROVED** [review]
Add description

**Review by Narayanbhat166 — APPROVED** [review]

**Pull request merged**